src/orion/benchmark/benchmark_client.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+:mod:`orion.benchmark.client` -- Benchmark client
+================================================================
+
+.. module:: client
+   :platform: Unix
+   :synopsis: Client to use Orion benchmark.
+
+"""
+import datetime
+import importlib
+
+from orion.benchmark import Benchmark, Study
+from orion.core.utils.exceptions import NoConfigurationError
+from orion.storage.base import get_storage, setup_storage
+
+
+def get_or_create_benchmark(name, algorithms=None, targets=None):
+    """
+    Create or get a benchmark object.
+
+    Parameters
+    ----------
+    name: str
+        Name of the benchmark
+    algorithms: list, optional
+        Algorithms used for benchmark, each algorithm can be a string or dict.
+    targets: list, optional
+        Targets for the benchmark, each target will be a dict with two keys.
+
+        assess: list
+            Assessment objects
+        task: list
+            Task objects
+
+    Returns
+    -------
+    An instance of `orion.benchmark.Benchmark`
+    """
+    setup_storage()
+    # fetch benchmark from db
+    db_config = _fetch_benchmark(name)
+
+    benchmark_id = None
+
+    if db_config:
+        benchmark_id, algorithms, targets = _resolve_db_config(db_config)
+
+    if not algorithms or not targets:
+        raise NoConfigurationError(
+            "Benchmark {} does not exist in DB, "
+            "algorithms and targets space was not defined.".format(name)
+        )
+
+    benchmark = _create_benchmark(name, algorithms, targets)
+
+    if benchmark_id is None:
+        # persist benchmark into db
+        _register_benchmark(benchmark)
+
+    return benchmark
+
+
+def _resolve_db_config(db_config):
+
+    benchmark_id = db_config["_id"]
+    algorithms = db_config["algorithms"]
+
+    obj_targets = []
+    str_targets = db_config["targets"]
+    for target in str_targets:
+        obj_target = {}
+
+        assessments = target["assess"]
+        obj_assessments = []
+        for assessment in assessments:
```

Comment:
That's a lot of things to process that we should probably be able to avoid by using the Factory instead.

Reply:
Can you be more specific about `Factory`? do you mean the `Factory design pattern` or similar stuff in the orion like below, or something else?
https://github.com/Epistimio/orion/blob/develop/src/orion/core/utils/__init__.py#L31

Reply:
Yeah I meant the Factory in Oríon, like it is being used for the algorithm: https://github.com/Epistimio/orion/blob/develop/src/orion/algo/base.py#L295